### PR TITLE
add logs api for pods

### PR DESCRIPTION
**Pull request by tornado404**
### PR types
New features

### PR changes
Others

### Describe
add logs api for pods

### CODE REVIEW

File: pkg/apiserver/controller/log/k8s_log.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package log
```
Comment (D0m021ng) [inline pkg/apiserver/controller/log/k8s_log.go]: rename file name.

File: pkg/apiserver/router/v1/log.go [hunk 2]
```diff
@@ -115,3 +128,151 @@ func (lr *LogRouter) getRunLog(writer http.ResponseWriter, request *http.Request
...
+		ctx.Logging().Errorln(err)
+		return logRequest, err
+	}
+	logRequest.ClusterInfo = clusterInfo
+
+	// validate jobID
+	jobID := request.URL.Query().Get(util.ParamKeyJobID)
+	if jobID != "" {
+		err := validateJobInfo(ctx, request, jobID)
+		if err != nil {
```
Comment (D0m021ng) [inline pkg/apiserver/router/v1/log.go]: Add log.

File: pkg/apiserver/router/v1/log.go [hunk 3]
```diff
@@ -115,3 +128,151 @@ func (lr *LogRouter) getRunLog(writer http.ResponseWriter, request *http.Request
...
+		err = fmt.Errorf("get k8s logs %s failed. err:%v", logRequest.Name, err)
+		ctx.Logging().Errorln(err)
+		common.RenderErrWithMessage(writer, ctx.RequestID, ctx.ErrorCode, err.Error())
+		return
+	}
+	common.Render(writer, http.StatusOK, &response)
+}
+
+func constructJobLogRequest(ctx *logger.RequestContext, request *http.Request) (runLog.GetMixedLogRequest, error) {
+	log.Infof("constructJobLogRequest, request: %v", request)
```
Comment (D0m021ng) [inline pkg/apiserver/router/v1/log.go]: `ctx.Logging().Infof("xxx")`

File: pkg/apiserver/router/v1/log.go [hunk 4]
```diff
@@ -115,3 +128,151 @@ func (lr *LogRouter) getRunLog(writer http.ResponseWriter, request *http.Request
...
+// @Success 200 {object} schema.JobLogInfo "日志详情"
+// @Failure 400 {object} common.ErrorResponse "400"
+// @Failure 500 {object} common.ErrorResponse "500"
+// @Router /log/job [GET]
+func (lr *LogRouter) getJobLog(writer http.ResponseWriter, request *http.Request) {
+	ctx := common.GetRequestContext(request)
+	log.Infof("get resource logs")
+	logRequest, err := constructJobLogRequest(&ctx, request)
+	if err != nil {
+		log.Errorf("parse job log request failed, err: %v", err)
```
Comment (D0m021ng) [inline pkg/apiserver/router/v1/log.go]: `ctx.Logging().Errorf("xxx")`

File: pkg/apiserver/router/v1/log.go [hunk 5]
```diff
@@ -115,3 +128,151 @@ func (lr *LogRouter) getRunLog(writer http.ResponseWriter, request *http.Request
...
+		ctx.ErrorCode = common.InvalidURI
+		common.RenderErrWithMessage(writer, ctx.RequestID, ctx.ErrorCode, err.Error())
+		return
+	}
+
+	var response schema.JobLogInfo
+	jobID := request.URL.Query().Get(util.ParamKeyJobID)
+	if jobID == "" {
+		log.Debugf("get kubernetes resource logs")
```
Comment (D0m021ng) [inline pkg/apiserver/router/v1/log.go]: `ctx.Logging().Debugf(""xx)`

File: pkg/apiserver/router/v1/log.go [hunk 6]
```diff
@@ -115,3 +128,151 @@ func (lr *LogRouter) getRunLog(writer http.ResponseWriter, request *http.Request
...
+		response, err = runLog.GetKubernetesResourceLogs(&ctx, logRequest)
+	} else {
+		log.Debugf("get paddleflow job %s logs", jobID)
```
Comment (D0m021ng) [inline pkg/apiserver/router/v1/log.go]: `ctx.Logging().Debugf("xx")`

**Review by luoyuedong — APPROVED** [review]

File: pkg/apiserver/router/v1/log.go (hunk 2 above)
Comment (tornado404) [reply pkg/apiserver/router/v1/log.go]: ok

File: pkg/apiserver/router/v1/log.go (hunk 3 above)
Comment (tornado404) [reply pkg/apiserver/router/v1/log.go]: done

File: pkg/apiserver/controller/log/k8s_log.go (hunk 1 above)
Comment (tornado404) [reply pkg/apiserver/controller/log/k8s_log.go]: done

File: pkg/apiserver/router/v1/log.go (hunk 4 above)
Comment (tornado404) [reply pkg/apiserver/router/v1/log.go]: done

File: pkg/apiserver/router/v1/log.go (hunk 6 above)
Comment (tornado404) [reply pkg/apiserver/router/v1/log.go]: done

File: pkg/apiserver/router/v1/log.go (hunk 5 above)
Comment (tornado404) [reply pkg/apiserver/router/v1/log.go]: done

File: pkg/apiserver/controller/log/cluster_log.go [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+		ctx.ErrorCode = common.InvalidArguments
+		ctx.Logging().Errorln(err)
+		return schema.JobLogInfo{}, err
+	}
+	return schema.JobLogInfo{}, nil
+}
+
+// GetLogs return mixed logs
+func GetLogs(ctx *logger.RequestContext, request GetMixedLogRequest) (schema.JobLogInfo, error) {
+	log.Debugf("Get mixed logs by request: %v", request)
```
Comment (D0m021ng) [inline pkg/apiserver/controller/log/cluster_log.go]: `ctx.Logging().Debugf("xx")`

File: pkg/apiserver/controller/log/cluster_log.go [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+		ResourceType:   request.ResourceType,
+		Framework:      request.Framework,
+		LineLimit:      request.LineLimit,
+		SizeLimit:      request.SizeLimit,
+		IsReadFromTail: request.IsReadFromTail,
+	}
+	response, err := runtimeSvc.GetLog(schema.JobLogRequest{}, schemaReq)
+	if err != nil {
+		err = fmt.Errorf("get mixed logs failed. error: %v", err)
+		ctx.ErrorCode = common.InvalidArguments
```
Comment (D0m021ng) [inline pkg/apiserver/controller/log/cluster_log.go]: `InvalidArguments` ?

File: pkg/apiserver/controller/log/cluster_log.go (hunk 7 above)
Comment (tornado404) [reply pkg/apiserver/controller/log/cluster_log.go]: done

File: pkg/apiserver/controller/log/cluster_log.go (hunk 8 above)
Comment (tornado404) [reply pkg/apiserver/controller/log/cluster_log.go]: InternalError

**Review by D0m021ng — APPROVED** [review]

**Pull request merged**